src/hooks/useOpenLoanHandler.tsx

```diff
@@ -106,212 +116,211 @@ export const useOpenLoanHandler = () => {
-      let collateralType = data.collateralType
-      let accountAddress = accountInfo.address ? accountInfo.address : ''
-
-      try {
-        let erc20
-        // TODO: adjust cases with new state, would probably want to ensureAllowance onChange of input
-        switch (collateralType) {
-          case CollateralType.LPToken:
-            // TODO: currently no way to know get uniPair without factory
-            // can't get it from position because there's no position yet
-            break
-          case CollateralType.Token0:
-            erc20 = new ethers.Contract(token0.address, IERC20.abi, provider)
-            await ensureAllowance(accountAddress, erc20, amt0BN, token0.decimals, token0.symbol)
-            break
-          case CollateralType.Token1:
-            // switch the amounts because amount comes from first input
-            amt1BN = amt0BN
-            amt0BN = BigNumber.from(0)
-            erc20 = new ethers.Contract(token1.address, IERC20.abi, provider)
-            await ensureAllowance(accountAddress, erc20, amt1BN, token1.decimals, token1.symbol)
-            break
-          case CollateralType.Both:
-            let erc20Token0 = new ethers.Contract(token0.address, IERC20.abi, provider)
-            let erc20Token1 = new ethers.Contract(token1.address, IERC20.abi, provider)
-            await ensureAllowance(accountAddress, erc20Token0, amt0BN, token0.decimals, token0.symbol).then(() =>
-              ensureAllowance(accountAddress, erc20Token1, amt1BN, token1.decimals, token1.symbol)
-            )
-            break
-          default:
-            notifyError('Invalid collateral type.')
-            return
-        }
-        // TODO: wait for contract to handle collateral
-        let tx = await posManager.openPosition(
-          token0.address,
-          token1.address,
-          amt0BN,
-          amt1BN,
-          loanAmtBN,
-          accountAddress
-        )
-        let loading = notifyLoading('Waiting for block confirmation')
-        let receipt = await tx.wait()
-        if (receipt.status == 1) {
-          notifyDismiss(loading)
-          notifySuccess('Position opened successfully.')
-          return
-        }
-        notifyError('Open position was unsuccessful.')
-      } catch (e) {
-        if (typeof e === 'string') {
-          notifyError(e)
-        } else if (e instanceof Error) {
-          notifyError(e.message)
+    if (!peripheryPosManager) {
+      await getPosMgr()
+    }
+    if (peripheryPosManager) {
+      let loan = await createLoan()
+      if (loan) {
+        const { args } = loan.events[1]
+        let tokenId = args.tokenId
+        console.log(tokenId)
+        if (tokenId) {
+          let collateral = await increaseCollateral(tokenId)
+          console.log(collateral)
+          if (collateral) {
+            let borrow = await borrowLiquidity(tokenId)
+            console.log(borrow)
+            if (borrow) {
```

Comment:
I feel there is too much nesting here, hard for readability, can it be de-nested?

Reply:
Fixed